salesforce-composite/2.7/modules/ROOT/pages/edits/salesforce-composite-connector-examples.adoc

```diff
@@ -0,0 +1,469 @@
+
+* HTTP *Listener*
++
+Accepts data from HTTP requests
+* *Transform Message*
++
+Specifies name for object
+* *Pre create*
++
+Provides metadata for the object that is going to be created and generates a subrequest used in execution
+* *Transform Message*
++
+Specifies values for fields
+* *Pre query*
++
+Provides metadata for the object that is going to be queried and generates a subrequest used in execution
+* *Transform Message*
++
+Passes output from *Pre query* operation
+* *Execute composite batch*
++
+Sends all subrequests in one batch
+* *Transform Message*
++
+Outputs the results in JSON format
++
+.The flow for this example includes an HTTP Listener component, Transform components, and multiple Salesfore Composite operations.
+image::salesforce-composite-prequery-flow.png[Pre-Query flow in Studio]
+
+=== Add an HTTP Listener
+
+The HTTP *Listener* source operation initiates the flow when it detects an event in the `executeFlow` path:
+
+. Create a new Mule project in Studio.
+. In the *Mule Palette* view, search for *HTTP* and select the *Listener* operation.
+. Drag the *Listener* operation onto the canvas.
+. In the Listener properties window, set the *Path* field to `/executeFlow`.
+
+=== Add the First Transform Message Component
+
+This *Transform Message* component specifies a name for the created object and passes the name to the *Pre create* operation:
+
+. In the *Mule Palette* view, search for *Transform Message*.
+. Drag the *Transform Message* component onto the canvas, to the right of *Listener*.
+. Enter the name `MyNewAccount`:
++
+[source,dataweave,linenums]
+----
+%dw 2.0
+%output application/java
+---
+{
+   Name: "MyNewAccount"
+}
+----
+
+=== Add the Pre Create Operation
+
+The *Pre create* operation provides metadata for creating an object called `NewAccount` and generates a subrequest for the `executeCompositeBatch` operation:
+
+. In the *Mule Palette* view, search for *Salesforce* and select the Salesforce Composite *Pre create* operation.
+. Drag the *Pre create* operation to the right of *Transform Message*.
+. Click the green plus icon (*+*) to the right of the *Connector configuration* field to access the Salesforce Composite global element configuration fields.
+. Complete the fields using variables, as follows:
+.. In the *Pre create* properties window, enter `Account` in the *Type* field.
+.. In the properties file `src/main/mule/mule-app.properties`, declare and enter values for the variables in the global element.
+
++
+.Complete fields in the General section to create a reusable global element for Salesforce Composite Connector.
+image::salesforce-composite-global-config.png[Salesforce Composite Connector global element configuration]
++
+
+=== Add the Second Transform Message Component
+
+This *Transform Message* component specifies values for the *ID* and *Type* fields, and then passes these values to the *Pre query* operation:
```

Comment:
I suggest replacing "this" with the specific count: the first, the second, the third. This helps locate the user in the flow.
GLOBAL: I have not marked each instance.